llmfoundry/models/layers/blocks.py

```diff
@@ -198,7 +198,7 @@ def forward(
                 m = self.norm_2(x)
 
         n = self.apply_ffn(attention_mask, m)
-        x = x + self.resid_ffn_dropout(n)
+        x = x + self.resid_ffn_dropout(n).to(x.device)
```

Comment:
the fact that this one is needed is scary...

Reply:
Discussed offline. Modified the code accordingly.

Reply:
why is this needed actually?